pkg/rpctype/rpctype.go

```diff
@@ -25,7 +25,8 @@ type RPCCandidate struct {
 }
 
 type ConnectArgs struct {
-	Name string
+	Name        string
+	MachineInfo string
```

Comment:
I think it's better to use `[]byte` because it may be large.

Reply:
Got it. But I am not sure why using `[]btye` will be better because I thought a `string` is just an immutable `[]byte` in go? Is it because passing `[]byte` to RPC will be more efficient? Thanks!

Reply:
I expect that we will build this blob in some more complex way -- by appending multiple files and maybe doing some transformations. It will be more efficient on a []byte. For strings any modifications (append/replace 1 byte) will result in full copy and tons of garbage generated.
Overall, my feeling on API design is that string is more suitable for something that is literally a short one-line string, like a name, or email, or file name. While []byte is more suitable for larger, multi-line or binary blobs, say file contents, or a megabyte of kernel output, etc.
It's usually more efficient if any transformations are done on client/server side, but also feels more declarative.